### DISCUSSION TITLE
Linux中，如何修改Certbot默认的证书路径权限，直接使用，而不需要像入门教程中那样安装证书到另外的路径

### DISCUSSION THREAD

**Discussion by jbbzone** [XTLS/Xray-core#3198] [Q&A]
我是根据入门教程中一步步操作的，在配置HTTPS证书的环节，遇到问题了：
官方教程中说是考虑新手不了解权限相关的操作，所以单独创建了一个路径来部署证书。官方教程中也说到Xray是使用nobody这个账号允许的。因为证书已经部署在一个固定的路径中了，我觉得没有必要再重新安装到另外一个路径中。因为我也不算完全的Linux新手，基本的权限等操作都还是会的。所以，直接使用了Certbot部署的证书路径，结果启动Xray报错，提示

`infra/conf: failed to parse certificate > open /etc/letsencrypt/live/x.ckg.gay/fullchain.pem: perm`

我用的Certbot部署HTTPS证书，证书的部署路径是：
/etc/letsencrypt/live/x.com/fullchain.pem
/etc/letsencrypt/live/x.com/privkey.pem

在网上查寻以下链接中类似问题的说明，跟着教程将/etc/letsencrypt/live/x.com/路径和里面的证书文件，全部设置权限644，或者将用户改成nobody，对应用户组改成nogroup，之后的报错仍然是这样的。
https://github.com/v2fly/fhs-install-v2ray/wiki/Insufficient-permissions-when-using-certificates-zh-Hans-CN

我不清楚，我是哪里姿势不对。为什么不行，我就是不想因为权限这一关过不去，就单独设置一个安装证书的自动脚本，重新创建一个路径来部署证书。

**Comment by chika0801** [comment]
/测试一下Xray安装时 https://github.com/XTLS/Xray-install

`bash -c "$(curl -L https://github.com/XTLS/Xray-install/raw/main/install-release.sh)" @ install -u root`

用root用户安，看解决不。

或者不用root用户安时，把

将 chika.example.com 的证书安装到 /etc/ssl/private 目录
cp /etc/letsencrypt/archive/*/fullchain*.pem /etc/ssl/private/fullchain.cer
cp /etc/letsencrypt/archive/*/privkey*.pem /etc/ssl/private/private.key

设置证书权限，配合Xray服务端配置文件
chown -R nobody:nogroup /etc/ssl/private
chmod -R 0644 /etc/ssl/private/*

类似这样操作一下 文件是0644，SSL的文件是nobody nogroup 
https://github.com/chika0801/Xray-install/blob/main/certbot.md

**Comment by chika0801** [comment]
x.ckg.gay 这文件夹，看了下好像是要0700，用户和组都是nobody nogroup，你自己测试下
